### ISSUE TITLE
Invalid character [44] search error in Opengrok

### ISSUE THREAD

**Issue by ranjantomtom** [OpenGrok/OpenGrok#1182]
We have installed latest version of tomcat (8) and java 8 on that OpenGrok is running but unfortunately getting error while selecting multiple repositories/projects and put some text in full search and click button search.

Below is the error details:    

An invalid character [44] was present in the Cookie value

java.lang.IllegalArgumentException: An invalid character [44] was present in the Cookie value
	at org.apache.tomcat.util.http.Rfc6265CookieProcessor.validateCookieValue(Rfc6265CookieProcessor.java:160)
	at org.apache.tomcat.util.http.Rfc6265CookieProcessor.generateHeader(Rfc6265CookieProcessor.java:109)
	at org.apache.catalina.connector.Response.generateCookieString(Response.java:989)
	at org.apache.catalina.connector.Response.addCookie(Response.java:937)
	at org.apache.catalina.connector.ResponseFacade.addCookie(ResponseFacade.java:386)
	at org.apache.jsp.search_jsp._jspService(search_jsp.java:211)
	at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:443)
	at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:385)
	at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:329)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:230)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:198)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:108)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:632)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:620)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:349)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:1110)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:785)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1425)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:745)

**Comment by vladak** [comment]
Please use the most recent version - 0.12.1.6 has a fix for this.

**Comment by ranjantomtom** [comment]
We already have opengrok-0.12.1.5.
Further, the same version working smooth on one centos server and failing on centos server.

**Comment by vladak** [comment]
0.12.1.6 was released just yesterday to address this very problem.

**Comment by ramgarden** [comment]
We've had this issue ever since we installed it too.  I don't know what version we have but it's definitely not the latest.  We'll update it now and check.

**Comment by ranjantomtom** [comment]
Hi Vladak,

Thank you for information.
Wanted to confirm if we need to replace source war file under tomcat webapps or just opengrok bin containing (Opengrok executable) file should be ok or we need to replace both the directory?

**Comment by vladak** [comment]
Replacing the war file should be enough.

**Comment by pankaj161993** [comment]
can anyone send the path to download the OpenGrok 0.12.1.6

**Comment by ranjantomtom** [comment]
You can download from here
https://github.com/OpenGrok/OpenGrok/releases

**Comment by pankaj161993** [comment]
thanks Rajiv Ranjan

**Comment by aez899** [comment]
So, there is still problem with newer releases with non-URLEncoded cookies which break recent Tomcat? I used Opengrok 1.0, Tomcat 9 and still this error:

An invalid character [44] was present in the Cookie value

java.lang.IllegalArgumentException: An invalid character [44] was present in the Cookie value

at org.apache.tomcat.util.http.Rfc6265CookieProcessor.validateCookieValue(Rfc6265CookieProcessor.java:182)
at org.apache.tomcat.util.http.Rfc6265CookieProcessor.generateHeader(Rfc6265CookieProcessor.java:115)
at org.apache.catalina.connector.Response.generateCookieString(Response.java:1009)
at org.apache.catalina.connector.Response.addCookie(Response.java:961)
at org.apache.catalina.connector.ResponseFacade.addCookie(ResponseFacade.java:383)
at org.apache.jsp.search_jsp._jspService(search_jsp.java:214)
at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:742)
at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:443)
at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:385)
at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:329)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:742)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199)
at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:592)
at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140)
at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:80)
at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:624)
at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342)
at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:498)
at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:796)
at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1366)
at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
at java.lang.Thread.run(Unknown Source)

**Comment by vladak** [comment]
Did you perhaps upgrade recently from older release and still have original cookies ?

**Comment by aez899** [comment]
No, I installed clean and fresh installation. Should I install 0.12.1.6 and upgrade it to 1.0?

**Comment by vladak** [comment]
No, It matters if there were any opengrok cookies left in your browser.

**Comment by vladak** [comment]
Most browsers these days have an option to selectively remove cookies for given site. This should be done after the upgrade for all users who accesss your deployment and encountered this problem.

**Comment by aez899** [comment]
Hmm ok, we had opengrok 0.11 some time ago, maybe I have old cookies in browser. I'll check this.

**Comment by bobbyhubbard** [comment]
We too are seeing the "An invalid character [44] was present in the Cookie value" with 0.12.1.6. Selecting a single project there is no problem. selecting 2 or more projects (including select all) results in the invalid character error.

It looks like it coming from the response cookie NOT the request cookies which is why clearing cookies has no affect. Maybe the delimiter that is used when multiple projects are selected or something? I haven't actually looked at the code yet.

```
An invalid character [44] was present in the Cookie value
	at org.apache.tomcat.util.http.Rfc6265CookieProcessor.validateCookieValue(Rfc6265CookieProcessor.java:182)
	at org.apache.tomcat.util.http.Rfc6265CookieProcessor.generateHeader(Rfc6265CookieProcessor.java:115)
	at org.apache.catalina.connector.Response.generateCookieString(Response.java:1019)
...
```

**Comment by vladak** [comment]
Time to upgrade to 1.0 or 1.1-rcX I'd say.

**Comment by bobbyhubbard** [comment]
SOLVED  The search.jsp in the war is not the same as what was in the tar.gz because we were downloading the wrong file. :)

So I checked the package. My script calls this url:
`https://github.com/OpenGrok/OpenGrok/files/213268/opengrok-0.12.1.6.tar.gz`

Instead of getting the actual path from the release notes, we simply incremented the version to 0.12.1.6 on the path from the 0.12.1.5 release. Since there was no 404, we assumed all was good. However, Github actually responds with:
`Content-Disposition →attachment;filename=opengrok-0.12.1.5.tar.gz`

If you use the link referenced on the actual release...  
`https://github.com/OpenGrok/OpenGrok/files/467358/opengrok-0.12.1.6.tar.gz.zip`
